src/addrman.h

```diff
@@ -723,14 +723,20 @@ friend class CAddrManTest;
         return addrRet;
     }
 
-    //! Return a bunch of addresses, selected at random.
-    std::vector<CAddress> GetAddr(size_t max_addresses, size_t max_pct)
+    /**
+     * Return all or many randomly selected addresses, optionally by network.
+     *
+     * @param[in] max_addresses  Maximum number of addresses to return (0 = all).
+     * @param[in] max_pct        Maximum percentage of addresses to return (0 = all).
+     * @param[in] network        Select only addresses of this network (nullopt = all).
+     */
+    std::vector<CAddress> GetAddr(size_t max_addresses, size_t max_pct, std::optional<Network> network)
```

Comment:
8942494e8c5eff8a3e268fe830868faf8ce19f9f

Why not `network = std::nullopt`? It would avoid changes in call sites.

Reply:
Yes, that is what the PR initially did, as it was a smaller diff and I tend to start with the smaller change. It is now passed explicitly in reponse to review feedback. I agree that explicit is good too.

Reply:
You can dismiss my comment, just checked @vasild comment https://github.com/bitcoin/bitcoin/pull/21843#discussion_r625554482.